src/components/DropdownSelect.js

```diff
@@ -89,7 +89,9 @@ export const DropdownSelect = ({
           let updatedData = [...data];
 
           if (multipleSelect) {
-            updatedData[index].selected = !updatedData[index].selected;
+            updatedData.map((entry) => {
+              if (entry.value === value) entry.selected = !entry.selected;
+            });
```

Comment:
nice, so this is the same as for not multiple, right?

```
updatedData = updatedData.map((entry) => ({
  ...entry,
  selected: entry.value === value
}));
```

i will test it and then reduce duplication before merge.

Reply:
in theory it looks the same. but i tried both. it works better that way. it will be much better if you find a more optimized method.

Reply:
no not at all. it needs to be different.

Reply:
but i missed the returned value updating `updatedData`, which we pass in `setData(updatedData);` a few lines later. so i added that part. it seemed that mutating `updatedData` implicitly also worked, but isn't that readable and understandable.